### DISCUSSION TITLE
How does Paru handle ASP support?

### DISCUSSION THREAD

**Discussion by ThisNekoGuy** [Morganamilo/paru#780] [Q&A] [answered]
I've seen that ASP is an opt-depend but there's no reference for usage at all in the man page and other references of information for Paru that would be expected to find relevant info on this. :/

**Comment by Morganamilo** [comment]
I guess the AUR strips the reason string from depends but it's there under pacman -Qi paru `asp: downloading repo pkgbuilds`

**Reply by ThisNekoGuy** [reply]
I mean... I know what asp _is_ and why it's used but I've never seen how it's integrated with Paru?
Like... how does Paru utilize and manage repo pkgbuilds via asp? Or does it just not?

**Reply by ThisNekoGuy** [reply]
@Morganamilo I had assumed that Paru interacted with asp more directly... Kind of like it does with the AUR Lol
That's why I was asking

**Reply by parkerlreed** [reply]
It's using asp directly to clone it, same as if you ran `asp checkout blah`.

**Reply by ThisNekoGuy** [reply]
So it does, then? But there's no explanation on how to make it do so anywhere, is what I'm getting at :/
There's no parameter for it and telling Paru to grab stuff from the repos normally just downloads the packages, so... clarification would be great Lol

**Reply by parkerlreed** [reply]
... https://github.com/Morganamilo/paru/blob/master/man/paru.8#L33

**Reply by ThisNekoGuy** [reply]
I... don't recall ever seeing that :v
Thanks for pointing that out; I've looked over the man page many times, so I'm not sure why this ever went over my head. :/

**Reply by ThisNekoGuy** [reply]
@Morganamilo  @parkerlreed Actually, I just tried that and all it does is download it...
I mean... that's not incorrect, it does what it says, I was just hoping it would treat `asp` more like it does the AUR and build what I ask... :/
What it does currently is so simple that I don't understand why I would use it over just using asp myself :/
Asp is cool, don't get me wrong, but the only reason I don't use it is because:

1. Managing a full compiled system build manually with asp is a major hassle (I've tried it by building a script before) - usually because of dependency orders
2. Certain packages (mostly 32-bit ones) have certain compiler setups that completely interfere with `makepkg.conf`

If I _were_ to suggest a feature to better support this with Paru... would be to treat `asp` like a secondary "AUR" and parse 32-bit package names and ensure that the proper 32-bit flags are appended... The 32-bit packages also tend to exclusively favor GCC over Clang and attempt to sort of handle this same issue by shoving/appending the 32-bit flag at the end of the compiler value to avoid touching the compiler flags (and this works for some odd reason) but I'm not exactly sure how else to deal with the 32-bit problem here concerning compiler choice... In my previous forays with `asp`, these 32-bit builds often reject clang for one reason or another. :/

Not sure if this matters to you at all or not, but I hope this discussion sheds some light on that.

**Reply by Morganamilo** [reply]
I'm not sure what your huge confusion is. asp is a dependency because paru needs it to do some things. Usually dependencies are used privately. Paru also optdedepends on bat and devtools. These are uses these internally to get stuff done. There's no need for documentation for how paru uses these beyond the one line in the optdepends.

**Reply by ThisNekoGuy** [reply]
@Morganamilo I'm not confused anymore, I was making a suggestion about the current asp integration

**Comment by ThisNekoGuy** [comment, marked as answer]
https://github.com/Morganamilo/paru/discussions/780#discussioncomment-3034330
> <**parkerlreed**> ... https://github.com/Morganamilo/paru/blob/master/man/paru.8#L33
